test/gtest/ucm/malloc_hook.cc

```diff
@@ -756,6 +757,21 @@ UCS_TEST_F(malloc_hook_cplusplus, remap_override) {
         EXPECT_EQ(size, m_unmapped_size);
     }
 
+    /* 8. sbrk call */
+    {
+        if (!RUNNING_ON_VALGRIND) {
+            /* valgrind failed when sbrk is called directly */
+            m_mapped_size = m_unmapped_size = 0;
+
+            /* call sbrk +/- closely to prevent heap destruction */
```

Comment:
seems suspicious..

Reply:
actually not: in case if between sbrk(+) and sbrk(-) malloc is called, and malloc calls sbrk(+), then sbrk(-) is called from test - heap will be destroyed completely, it was reason why mallocs can't be called from SBRK event

Reply:
can this really happen in this test? are there additional threads calling malloc?
anyhow the comment should include the explanation above..

Reply:
comment in source was about "do not add any evaluations between sbrk+/sbrk-", not threads. sbrk is not thread safe at all, it definitely break everything in case if multiple threads are calling it

I will update comment